Dockerfile

```diff
@@ -0,0 +1,46 @@
+FROM continuumio/miniconda3
+
+# if you forked pandas, you can pass in your own GitHub username to use your fork
+# i.e. gh_username=myname
+ARG gh_username=pandas-dev
+ARG pandas_home="/home/pandas"
+
+# Avoid warnings by switching to noninteractive
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Configure apt and install packages
+RUN apt-get update \
+    && apt-get -y install --no-install-recommends apt-utils dialog 2>&1 \
+    #
+    # Verify git, process tools, lsb-release (common in install instructions for CLIs) installed
+    && apt-get -y install git iproute2 procps iproute2 lsb-release \
+    #
+    # Install C compilers (gcc not enough, so just went with build-essential which admittedly might be overkill),
+    # needed to build pandas C extensions
+    && apt-get -y install build-essential \
+    #
+    # cleanup
+    && apt-get autoremove -y \
+    && apt-get clean -y \
+    && rm -rf /var/lib/apt/lists/*
+
+# Switch back to dialog for any ad-hoc use of apt-get
+ENV DEBIAN_FRONTEND=dialog
+
+# Clone pandas repo
+RUN mkdir "$pandas_home" \
+    && git clone "https://github.com/$gh_username/pandas.git" "$pandas_home" \
+    && cd "$pandas_home" \
+    && git remote add upstream "https://github.com/pandas-dev/pandas.git"
```

Comment:
Can you pull from upstream master after this step? Should keep up with the latest changes

Reply:
just did

Reply:
Hmm don't see this in the diff. Should be a `git pull upstream master` in there

Reply:
Oh I thought you meant on my changes in the PR, lol, not in the DockerFile. got it

Reply:
I think @WillAyd is requesting that the Dockerfile has a `git pull upstream master`?

Though I'm not sure that would achieve the desired effect, since it only happens when the docker image is built. How often does that happen? How does a person starting up a container ensure they're working with a fresh copy of pandas?

Reply:
To be honest, the main use case is setting up a fresh env for some dev work, using this `DockerFile` and the `devcontainer.json`. Once you are using that env, if you use it for a while, it's up to the user to keep the code up to date with master. This is just intended for first time set up.

But the real value of this is that's it's dead simple to delete and create a new env in VS Online or with containers, so the usual use case is just to create a new env and not use an existing one for a while. 

So I guess this line is indeed unnecessary, and I should remove it.

Reply:
Let me know if you agree before I revert that commit

Reply:
I think either this should stay or you remove the build step later. Doesn't make sense to build an outdated version for development

Reply:
not sure I follow you there. this is the same steps as the instructions in the main docs. Even if we remove `git pull upstream master`, how is this an outdated version? We just cloned the repo 2 lines before this, like 5 seconds ago